### PR TITLE
fix: minio error while creating bucket if already exists

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

- Since for every upload, we are trying to create the bucket. In case of the bucket already exists, we try to check if the bucket exists or not using `minioClient.BucketExists` but this is returning a `permission error`.
- We particularly don't need an API call to verify if the bucket exists or not. We can suppress the error from the error code itself when we get a bucket creation error.

## Notion Ticket

https://www.notion.so/rudderstacks/Minio-data-retention-object-storage-validations-is-failing-b1c961c67337476a82566a43eaed243e?pvs=4

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: services/filemanager/miniomanager.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ func (manager *MinioManager) Upload(ctx context.Context, file *os.File, prefixes
 	defer cancel()
 
 	if err = minioClient.MakeBucket(ctx, manager.Config.Bucket, minio.MakeBucketOptions{Region: "us-east-1"}); err != nil {
-		exists, errBucketExists := minioClient.BucketExists(ctx, manager.Config.Bucket)
-		if !(errBucketExists == nil && exists) {
+		err = suppressMinorErrors(err)
```
Comment (lvrach) [inline services/filemanager/miniomanager.go:39]: [minor] not sure if wrapping error handling in a separate method is really necessary. The errors that we are ignoring here are specific to bucket creation, not sure if the function should be reused elsewhere

**Review by lvrach — APPROVED** [review]

**Review by cisse21 — APPROVED** [review]

**Review by saurav-malani — COMMENTED** [review]
Instead of directly trying to create a bucket and then handling the error, why don't we 1st check if the bucket exist or not. And only create if it doesn't exist?

**Comment by saurav-malani** [comment]
> Yeah, But it will still require 1 API call anyways. Why not just use the Create APi and suppress it in case of already exists?
Well, I believe preventing error in first place is better then handling it. 
Also, according to me creating of a bucket is like a `write` operation where as checking if it exists or not is like a `read` operation. And, read would have less overhead in comparison to write in terms of internal implementation.

**Comment by achettyiitr** [comment]
> > Yeah, But it will still require 1 API call anyways. Why not just use the Create APi and suppress it in case of already exists?
> 
> Well, I believe preventing error in first place is better then handling it. Also, according to me creating of a bucket is like a `write` operation where as checking if it exists or not is like a `read` operation. And, read would have less overhead in comparison to write in terms of internal implementation. In short, the overhead of both the api call would be quite different.

Make sense.

**Comment by achettyiitr** [comment]
What if we don't create a bucket at all? Ideally, it should be the customer's responsibility to create the bucket.

**Comment by saurav-malani** [comment]
> What if we don't create a bucket at all? Ideally, it should be the customer's responsibility to create the bucket.

well, I am not sure if we have clearly mentioned it somewhere that they themself need to create a bucket. But, if that is the case then your point make sense. But, still I believe at the end of the it will escalate to some issue when upload would fail and then we will be notified and CS team will get in touch with them to create it. So, I personally think creating on our own if it doesn't exist would be best for us. 😄

**Comment by achettyiitr** [comment]
I have reverted the Azure changes since it requires additional logic to check if the container exists or not using [storage.GetContainerReferenceFromSASURI().Exists()](https://github.com/Azure/azure-sdk-for-g) which requires a SASUri

**Comment by achettyiitr** [comment]
Also, it was not the scope of the PR.

**Review by saurav-malani — APPROVED** [review]

**Pull request merged**